pkg/kubectl/resource/result.go

```diff
@@ -228,6 +229,18 @@ func AsVersionedObject(infos []*Info, forceList bool, version unversioned.GroupV
 		}
 		object = converted
 	}
+
+	// validSpecifiedVersion resolves to true if the version passed to this function matches the
+	// version assigned to the converted object
+	actualVersion := object.GetObjectKind().GroupVersionKind()
+	if actualVersion.Version != version.Version {
+		defaultVersionInfo := ""
+		if len(actualVersion.Version) > 0 {
+			defaultVersionInfo = fmt.Sprintf("Defaulting to %q", actualVersion.Version)
+		}
+		glog.V(1).Infof(" info: the output version specified is invalid. %s\n", defaultVersionInfo)
```

Comment:
@ncdc Okay, updated this here https://github.com/kubernetes/kubernetes/pull/34763/commits/6fd704c174df46cae0a6d4a4b13f2ed76d358752